### ISSUE TITLE
Extra directories when packing from msbuild with %(RecursiveDir)

### ISSUE THREAD

**Issue by ellismg** [NuGet/Home#4818]
## Details about Problem

dotnet.exe --version (if appropriate):

```
.NET Command Line Tools (1.0.0)

Product Information:
 Version:            1.0.0
 Commit SHA-1 hash:  e53429feb4

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 RID:         ubuntu.14.04-x64
 Base Path:   /home/matell/.dotnet/sdk/1.0.0
```
## Detailed repro steps so we can see the same problem

This is reduced from a real project I am trying to convert to use the new NuGet pack stuff in the dotnet:  My project layout looks like this:

```
.
├── Class1.cs
├── Content
│   ├── A
│   │   └── A.txt
│   ├── B
│   │   └── B.txt
│   └── C
│       └── C.txt
└── test.csproj
```

I want to put everything under the Content folder into my nupkg, mirroring the directory layout (into the tools\net46 folder).

In the csproj, I have the following:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard1.4</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Content Include="Content/**/*">
      <PackagePath>tools\net46\%(RecursiveDir)</PackagePath>
    </Content>
  </ItemGroup>

</Project>
```

The finals paths of the content in the nuspec are "duplicated":

```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
  <metadata>
    <id>test</id>
    <version>1.0.0</version>
    <authors>test</authors>
    <owners>test</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <description>Package Description</description>
    <dependencies>
      <group targetFramework=".NETStandard1.4">
        <dependency id="NETStandard.Library" version="1.6.1" exclude="Build,Analyzers" />
      </group>
    </dependencies>
  </metadata>
  <files>
    <file src="/home/matell/tmp/test/bin/Debug/netstandard1.4/test.dll" target="lib/netstandard1.4/test.dll" />
    <file src="/home/matell/tmp/test/Content/A/A.txt" target="tools\net46\A/A/A.txt" />
    <file src="/home/matell/tmp/test/Content/B/B.txt" target="tools\net46\B/B/B.txt" />
    <file src="/home/matell/tmp/test/Content/C/C.txt" target="tools\net46\C/C/C.txt" />
  </files>
</package>
```

Note how `Content/A/A.txt` gets mapped to `tools\net46\A\A\A.txt` instead of `tools\net46\A\A.txt`

From the msbuild log it looks like stuff is being passed to the Pack Task correctly:

```
                   Task "PackTask" (TaskId:45)
                     Task Parameter:PackItem=/home/matell/tmp/test/test.csproj (TaskId:45)
                     Task Parameter:
                         PackageFiles=
                             Content/A/A.txt
                                     BuildAction=Content
                                     PackagePath=tools\net46\A/
                             Content/B/B.txt
                                     BuildAction=Content
                                     PackagePath=tools\net46\B/
                             Content/C/C.txt
                                     BuildAction=Content
                                     PackagePath=tools\net46\C/ (TaskId:45)
                     Task Parameter:PackageVersion=1.0.0 (TaskId:45)
                     Task Parameter:PackageId=test (TaskId:45)
                     Task Parameter:Authors=test (TaskId:45)
                     Task Parameter:Description=Package Description (TaskId:45)
                     Task Parameter:RequireLicenseAcceptance=False (TaskId:45)
                     Task Parameter:TargetPathsToAssemblies=/home/matell/tmp/test/bin/Debug/netstandard1.4/test.dll (TaskId:45)
                     Task Parameter:TargetFrameworks=netstandard1.4 (TaskId:45)
                     Task Parameter:AssemblyName=test (TaskId:45)
                     Task Parameter:PackageOutputPath=/home/matell/tmp/test/bin/Debug/ (TaskId:45)
                     Task Parameter:ContinuePackingAfterGeneratingNuspec=True (TaskId:45)
                     Task Parameter:NuspecOutputPath=obj/ (TaskId:45)
                     Task Parameter:IncludeBuildOutput=True (TaskId:45)
                     Task Parameter:BuildOutputFolder=lib (TaskId:45)
                     Task Parameter:
                         ContentTargetFolders=
                             content
                             contentFiles (TaskId:45)
                     Task Parameter:RestoreOutputPath=/home/matell/tmp/test/obj/ (TaskId:45)
                     Successfully created package '/home/matell/tmp/test/bin/Debug/test.1.0.0.nupkg'. (TaskId:45)
```

I see the same behavior on both windows and *nix.

It's possible I am doing the wrong gesture for what I want to do, but it seemed like this should have worked.

**Comment by ellismg** [comment]
Ahh, I jumped into the PackTask a bit and I think I figured out my problem.

The issue is here: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs#L352-L369

NuGet is adding the recursive dir for me already.  However, this only happens if the entry in PackagePath ends with Path.DirectorySeperatorChar, which means I'll need to define a MSBuild property if I want the pack to happen correctly if I build on either windows or *nix:  For example, I need to do this:

```
    <Content Include="Content/**/*">
      <PackagePath Condition="'$(OS)' == 'Windows_NT'">tools\net46\</PackagePath>
      <PackagePath Condition="'$(OS)' != 'Windows_NT'">tools/net46/</PackagePath>
    </Content>
```

And then things work as I would expect.

The behavior here is pretty surprising.  Given that I can use \ or / in PackagePath and things "just work" in the outputed nupkg, would be it be possible to just accept both \ and / when looking to see if Path.Combine should happen?

As is, writing some csproj that can be packed on multiple platforms with globing is tricky.
